01_fib/shahriys/main_test.go

```diff
@@ -0,0 +1,73 @@
+package main
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buffib bytes.Buffer
+	outfib = &buffib
+	main()
+	expected :=
+		`1
+1
+2
+3
+5
+8
+13
+1
+-1
+2
+-3
+5
+-8
+13
+`
+	actual := buffib.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main()")
+	}
+}
+func TestFibPos(t *testing.T) {
+	var buffib bytes.Buffer
+	outfib = &buffib
+
+	fib(7)
+
+	expected := `1
+1
+2
+3
+5
+8
+13
```

Comment:
I found when writing my expected result I was able to write 
```
expected := `1\n1\n2\n...`
```
rather than having my numbers hard against the left wall. Personal preference, but noting it here incase it influences your future tests. :)

Reply:
That's odd @Jake88, are you sure you don't mean to use `"` instead of backticks? ([see here](https://goplay.space/#drRsTZ-HVfK)). If you are testing with backticks, then the output of your function is wrong (each number is sparated by a literal `\n` instead of a newline).

As for whether to test agains quotes with `\n` or backticks with newlines, I think this comes down to preference, I personally prefer a string literal (backticks, what you currently have) as it actually looks like the output. I also don't find it too hard to see a bunch of `\n`s in a row in quotes, and know what the output should look like. I'll leave it up to you if you want to change it.